docs/configuration.rst

```diff
@@ -405,6 +405,61 @@ Use ``style_start`` and ``style_end`` like this::
       }
    ]
 
+.. _needs_filter_data:
+
+needs_filter_data
+~~~~~~~~~~~~~~~~~
+
+This option allows to custom define extra filter.
+
+Configuration example::
+
+   def custom_defined_func():
+       return "my_tag"
+
+   needs_filter_data = {
+       "current_variant": "project_x",
+       "sphinx_tag": custom_defined_func(),
+   }
+
+
+The defined ``needs_filter_data`` must be a dictionary. Its values can be a string variable or a custom defined 
+function. The function get execued during config loading and must return a string. 
+
+Hence, this is **not allowed**::
+
+   needs_filter_data = {
+       "sphinx_tag": custom_defined_func,  # () missing to execute the function
+   }
+
+
+The value of ``needs_filter_data`` will be used as filter string and can be very poweful togehter with 
+internal needs info to filter needs.
+
+The defined extra filter can be used like this::
+
+   .. needextend:: type == "req" and sphinx_tag in tags
+      :+tags: my_external_tag
+
+
+or if need has :ref:`needs_extra_options` defined like::
+
+   needs_extra_options = {
+       "variant": directives.unchanged,
+   }
+
+
+the defined extra filter can also be used like::
+
+   .. needextend:: type == "test" and variant == current_variant
+      :variant: my_external_variant
```

Comment:
Don't set variant here.
Better set ``:hide: True``

And for the filter ``variant != current_variant``

So all need objects get hidden, if the y are not defined for the current_variant